src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/_params.py

```diff
@@ -140,9 +144,23 @@ def __repr__(self):
     help='Specifies whether to enable zone redundancy',
     arg_type=get_three_state_flag())
 
+managed_instance_param_type = CLIArgumentType(
+    options_list=['--managed-instance', '--mi'],
+    help='Name of the Azure SQL managed instance.')
+
+storage_param_type = CLIArgumentType(
+    options_list=['--storage'],
+    type=SizeWithUnitConverter('GB', result_type=int, unit_map=dict(B=1.0 / (1024 * 1024 * 1024),
+                                                                    kB=1.0 / (1024 * 1024),
+                                                                    MB=1.0 / 1024,
+                                                                    GB=1,
+                                                                    TB=1024)),
```

Comment:
result_type and unit_map don't need to be explicitly specified here, right?

Reply:
Actually we have to, otherwise wrong dict will be used. In our case storage should be specified in GB and converted to it before sending the request

Reply:
Oh, I didn't see that the multipliers are different. Cool!